src/renderers/dom/shared/__tests__/inputValueTracking-test.js

```diff
@@ -153,4 +152,28 @@ describe('inputValueTracking', () => {
 
     expect(input.hasOwnProperty('value')).toBe(false);
   });
+
+  it('does not crash for nodes with custom value property', () => {
+    // https://github.com/facebook/react/issues/10196
+    try {
+      var originalCreateElement = document.createElement;
+      document.createElement = function() {
+        var node = originalCreateElement.apply(this, arguments);
+        Object.defineProperty(node, 'value', {
+          get() {},
+          set() {},
+        });
+        return node;
+      };
+      var div = document.createElement('div');
+      // Mount
+      var node = ReactDOM.render(<input type="text" />, div);
+      // Update
+      ReactDOM.render(<input type="text" />, div);
```

Comment:
`SimulateNative.change(node)` should work, if the event triggered it passed through the function. It isn't specifically gonna catch a failure in Fiber but it covers the all paths through `updateValueIfChanged`.